internals/overlord/servstate/handlers.go

```diff
@@ -340,16 +357,39 @@ func (s *serviceData) startInternal() error {
 	s.cmd = exec.Command(args[0], args[1:]...)
 	s.cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
 
-	// Copy environment to avoid updating original.
 	environment := make(map[string]string)
-	for k, v := range s.config.Environment {
-		environment[k] = v
+	if s.workload != nil {
+		maps.Copy(environment, s.workload.Environment)
 	}
+	// If both workload and service provides the same environment variable,
+	// the workload environment prevails.
```

Comment:
comment: I think we agreed the service environment must trump. Is that what the comment is saying ? It sounds inverted to me?